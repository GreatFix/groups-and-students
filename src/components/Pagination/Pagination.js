import React from 'react'
import classes from './Pagination.module.css'
import Button from '../Button/Button'
import PropTypes from 'prop-types'

const Pagination = (props) => {
  const numbers = []
  for (let i = 1; i <= props.size; i++) {
    numbers.push(i)
  }
  return (
    <div className={`${classes.Pagination} ${props.className}`}>
      {numbers.map((number) => {
        return (
          <Button
            key={Math.random()}
            onClick={() => {
              props.onClick(number)
            }}
          >
            {number}
          </Button>
        )
      })}
    </div>
  )
}

Pagination.propTypes = {
  className: PropTypes.string,
  size: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired,
}

Pagination.defaultProps = {
  className: '',
  size: 0,
  onClick: PropTypes.func,
}

export default Pagination
