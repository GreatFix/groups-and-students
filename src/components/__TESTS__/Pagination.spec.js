import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import userEvent from '@testing-library/user-event'
import Pagination from '../Pagination/Pagination'

const TEST_CLASS = 'TEST_CLASS'
const SIZE = 10

describe('Pagination', () => {
  test('renders buttons in the number of size', () => {
    const { container } = render(<Pagination size={SIZE} onClick={() => {}} />)
    const buttons = container.querySelectorAll('button')
    expect(buttons.length).toBe(SIZE)
  })

  test('gets a class', () => {
    const { container } = render(
      <Pagination size={SIZE} onClick={() => {}} className={TEST_CLASS} />
    )
    const pagination = container.querySelector('.Pagination')
    expect(pagination).toHaveClass(TEST_CLASS)
  })

  test('clicks on buttons', () => {
    const handleClick = jest.fn((number) => number)
    const { container } = render(<Pagination size={SIZE} onClick={handleClick} />)
    const buttons = container.querySelectorAll('button')

    userEvent.click(buttons[0])
    expect(handleClick).toHaveBeenCalledTimes(1)
    expect(handleClick.mock.results[0].value).toBe(1)

    userEvent.click(buttons[3])
    expect(handleClick).toHaveBeenCalledTimes(2)
    expect(handleClick.mock.results[1].value).toBe(4)

    userEvent.click(buttons[9])
    expect(handleClick).toHaveBeenCalledTimes(3)
    expect(handleClick.mock.results[2].value).toBe(10)
  })

  test('render properly', () => {
    const { container } = render(<Pagination size={SIZE} onClick={() => {}} />)
    expect(container).toMatchSnapshot()
  })
})
