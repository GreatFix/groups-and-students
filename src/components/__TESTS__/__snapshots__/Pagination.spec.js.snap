// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination render properly 1`] = `
<div>
  <div
    class="Pagination "
  >
    <button
      class="Button "
      style="background-color: rgb(0, 72, 186);"
      type="button"
    >
      1
    </button>
    <button
      class="Button "
      style="background-color: rgb(0, 72, 186);"
      type="button"
    >
      2
    </button>
    <button
      class="Button "
      style="background-color: rgb(0, 72, 186);"
      type="button"
    >
      3
    </button>
    <button
      class="Button "
      style="background-color: rgb(0, 72, 186);"
      type="button"
    >
      4
    </button>
    <button
      class="Button "
      style="background-color: rgb(0, 72, 186);"
      type="button"
    >
      5
    </button>
    <button
      class="Button "
      style="background-color: rgb(0, 72, 186);"
      type="button"
    >
      6
    </button>
    <button
      class="Button "
      style="background-color: rgb(0, 72, 186);"
      type="button"
    >
      7
    </button>
    <button
      class="Button "
      style="background-color: rgb(0, 72, 186);"
      type="button"
    >
      8
    </button>
    <button
      class="Button "
      style="background-color: rgb(0, 72, 186);"
      type="button"
    >
      9
    </button>
    <button
      class="Button "
      style="background-color: rgb(0, 72, 186);"
      type="button"
    >
      10
    </button>
  </div>
</div>
`;
